core/src/main/java/org/arquillian/smart/testing/configuration/ObjectMapper.java

```diff
@@ -0,0 +1,177 @@
+package org.arquillian.smart.testing.configuration;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+class ObjectMapper {
+
+    static <T extends ConfigurationSection> T mapToObject(Class<T> aClass, Map<String, Object> map) {
+        T instance;
+        try {
+            instance = aClass.newInstance();
+        } catch (InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException("Failed to create new instance of class " + aClass, e);
+        }
+
+        List<ConfigurationItem> configItems = instance.registerConfigurationItems();
+
+        Arrays.stream(aClass.getMethods()).filter(ObjectMapper::isSetter)
+            .forEach(method -> invokeMethodWithMappedValue(configItems, method, instance, map));
+
+        return instance;
+    }
+
+    private static <T> void invokeMethodWithMappedValue(List<ConfigurationItem> configItems, Method method, T instance,
```

Comment:
maybe break this method not to be so long - something like separating: https://gist.github.com/MatousJobanek/9e79ba640b3b98361c2afca283aa867a